proto/rpc.proto

```diff
@@ -53,6 +53,8 @@ enum ErrorReason
   METHOD_NOT_FOUND = 3;                         // Method not found on server
   RPC_ERROR = 4;                                // Rpc threw exception on server
   RPC_FAILED = 5;                               // Rpc failed on server
+  UNAUTHORIZED_REQUEST = 10;                    // Server received unauthorized request
```

Comment:
yes, we can keep things in sequence here but current code has 2 different blocks: Server-side errors and Client-side errors. If we do sequence it will lead to 3 blocks: Server-side errors, Client-side errors, Server-side new errors (?). Not ideal imho, but still doable.